### DISCUSSION TITLE
How to style React Native Navigation with NativeWind v4?

### DISCUSSION THREAD

**Discussion by denysoleksiienko** [nativewind/nativewind#989] [General]
```
<Tab.Navigator
  screenOptions={{
    headerShown: false,
    tabBarActiveTintColor: 'red',
    tabBarInactiveTintColor: 'black',
    tabBarLabelStyle: {
      fontSize: 12,
    },
    tabBarStyle: {
      backgroundColor: 'white',
      height: 40,
      borderTopWidth: 1,
      paddingVertical: 4,
    },
    tabBarItemStyle: { paddingBottom: 12 },
  }}
>
  <Tab.Screen
    name={'Home'}
    options={{
      tabBarStyle: {}
    }}
  >
</Tab.Navigator>
```

**Comment by joelpiccoli** [comment]
You can go with [remapProps](https://www.nativewind.dev/v4/api/remap-props). It's going to give you a little bit of work, but I believe it can be done that way.

**Reply by denysoleksiienko** [reply]
`remapProps` doesn't work at all

**Comment by shovel-kun** [comment]
`expo-router` is based off `react-navigation`,  so the following code should still work:

See:
- https://github.com/hoarder-app/hoarder/blob/300f3c5d0b661c430ad2f6b479b151ec65f14243/apps/mobile/components/navigation/tabs.tsx
- https://github.com/hoarder-app/hoarder/blob/300f3c5d0b661c430ad2f6b479b151ec65f14243/apps/mobile/components/navigation/stack.tsx

**Comment by ferntheplant** [comment]
I think the issue is that the underlying react-navigation `<BottomTabsBar />` component has lots of default styles applied to its internal `<View />` component. These default styles are then rendered in-line whereas the Nativewind `className` styles (using the interop strategy suggested above from the Hoarder app) are applied using normal CSS classes. The in-line styles will have priority over the className styles so your Nativewind styles may not show up correctly depending on how they interact with the defaults react-navigation chose.

I personally do not like the defaults so I instead provided my own `tabBarBackground` renderer and surrendered to writing the remaining styles for this navigator without Nativewind. I tried making a totally custom `Tabs` component using the [guide here](https://docs.expo.dev/router/advanced/custom-tabs/), but even the elements that are supposed to be highly customizable apply some default in-line styles using the Stylesheet API (see [here](https://github.com/expo/expo/blob/62757baa3cd3377e9fc7909aea2a4f3d4d716f18/packages/expo-router/src/ui/TabList.tsx#L26)) that override anything applied via `className`.

**Comment by danstepanov** [comment]
Closing this discussion, if this is still relevant, feel free to open a new one.
